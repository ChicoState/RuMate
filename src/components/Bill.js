import React from 'react';
import { View, Text, StyleSheet } from 'react-native';

const Bill = ({ name, value, due, payed, edit }) => {
  if (payed) {
    payed = styles.payed
  } else {
    payed = styles.neutral
  }
  if (!edit) {
    return (
      <View style={[styles.bill, payed]}>
        <Text style={styles.title}>{name}</Text>
        <View style={styles.detail}>
          <Text>Amount: ${value}</Text>
          <Text>Due by: {due}</Text>
        </View>
      </View>
    );
  } else {
    return (
      <View style={[styles.bill, payed]}>
        <Text style={styles.title}>{name}</Text>
        <View style={styles.detail}>
          <Text>Amount: ${value}</Text>
          <Text>Due by: {due}</Text>
          <Text>Editing</Text>
        </View>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  neutral: {
    backgroundColor: 'white'
  },
  payed: {
    backgroundColor: 'green'
  },
  title: {
    fontSize: 18
  },
  bill: {
    paddingVertical: 20,
    paddingLeft: 10,
    borderBottomWidth: 1,
    borderBottomColor: '#000',
  },
  detail: {
    paddingLeft: 10
  }
});

export default Bill;