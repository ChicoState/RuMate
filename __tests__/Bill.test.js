import React from 'react';
import renderer from 'react-test-renderer';

import Bill from '../src/components/Bill';

describe('<Bill />', () => {

  it('has 1 child', () => {
    const tree = renderer.create(<Bill />).toJSON();
    console.log(tree.children.length)
    expect(tree.children.length).toBe(2);
  });
  
  it('renders correctly', () => {
    const tree = renderer.create(<Bill />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('checks payed', () => {
    const tree = renderer.create(
      <Bill payed={true}/>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it ('checks editing', () => {
    const tree = renderer.create(
      <Bill edit={true} />
    ).toJSON();
  })
  
});

