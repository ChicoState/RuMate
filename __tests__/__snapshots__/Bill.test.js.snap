// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Bill /> checks payed 1`] = `
<View
  style={
    Array [
      Object {
        "borderBottomColor": "#000",
        "borderBottomWidth": 1,
        "paddingLeft": 10,
        "paddingVertical": 20,
      },
      Object {
        "backgroundColor": "green",
      },
    ]
  }
>
  <Text
    style={
      Object {
        "fontSize": 18,
      }
    }
  />
  <View
    style={
      Object {
        "paddingLeft": 10,
      }
    }
  >
    <Text>
      Amount: $
    </Text>
    <Text>
      Due by: 
    </Text>
  </View>
</View>
`;

exports[`<Bill /> renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "borderBottomColor": "#000",
        "borderBottomWidth": 1,
        "paddingLeft": 10,
        "paddingVertical": 20,
      },
      Object {
        "backgroundColor": "white",
      },
    ]
  }
>
  <Text
    style={
      Object {
        "fontSize": 18,
      }
    }
  />
  <View
    style={
      Object {
        "paddingLeft": 10,
      }
    }
  >
    <Text>
      Amount: $
    </Text>
    <Text>
      Due by: 
    </Text>
  </View>
</View>
`;
